core/domain/role_services.py

```diff
@@ -131,3 +142,69 @@ def get_all_actions(role):
         role_actions.extend(get_all_actions(parent_role))
 
     return list(set(role_actions))
+
+
+def get_updatable_roles():
+    """Returns roles to which a user's role can be updated by admin interface.
+
+    Returns:
+        dict. A dict of all roles that a user can be given by admin.
+            key -> roles in format as stored in backed.
+            value -> roles in format to show users.
+    """
+    updatable_roles = {}
+    for role in PARENT_ROLES:
+        if role not in [feconf.ROLE_GUEST]:
+            updatable_roles[role] = get_human_readable_role(role)
+    return updatable_roles
+
+
+def get_viewable_roles():
+    """Returns roles by which users can be searched in admin interface.
+
+    Returns:
+        dict. A dict of all roles by which user's can be searched.
+            key -> roles in format as stored in backend.
+            value -> roles in format to show users.
+    """
+    viewable_roles = {}
+    for role in PARENT_ROLES:
+        if role not in [feconf.ROLE_EXPLORATION_EDITOR, feconf.ROLE_GUEST]:
+            viewable_roles[role] = get_human_readable_role(role)
+    return viewable_roles
+
+
+def get_role_graph_data():
+    """Returns dict for displaying roles graph.
+
+    Returns:
+        dict. A dict containing data in following format:
+        {
+            finalStateIds: list(str). Roles which have no child.
```

Comment:
I think they are also use to traverse graph in StateGraphLayoutService. Since, for a general graph, the notion that no inputs = initial node should not be valid (as a node may have edge to init node, that's why they are there in StateGraphLayoutService.